### DISCUSSION TITLE
Node Classification in Inductive Learning Tasks

### DISCUSSION THREAD

**Discussion by hengdezhu** [danielegrattarola/spektral#149] [Q&A]
Hi @danielegrattarola,

Given a graph G(V, E), how to do node classification in inductive learning tasks? For example, we contruct a graph for the network training. When new nodes come, how to contruct test dataset and evaluate them?

Thank you

**Comment by danielegrattarola** [comment]
Hi, 
You simply introduce the new nodes into your graph and run the model in the same way as you did before. 

If the new nodes represent a different graph (i.e., there are no edges between the training nodes and the test nodes) then you can simply run the prediction on the new graph, ignoring the previous data. Otherwise, you need to feed the full graph to the model. 

Hope this helps.
Cheers
